src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -13835,6 +13835,18 @@ private static Vector128<T> ElementWiseSelect<T>(Vector128<T> mask, Vector128<T>
                 if (typeof(T) == typeof(long)) return Sse41.BlendVariable(left.AsInt64(), right.AsInt64(), (~mask).AsInt64()).As<long, T>();
                 if (typeof(T) == typeof(float)) return Sse41.BlendVariable(left.AsSingle(), right.AsSingle(), (~mask).AsSingle()).As<float, T>();
                 if (typeof(T) == typeof(double)) return Sse41.BlendVariable(left.AsDouble(), right.AsDouble(), (~mask).AsDouble()).As<double, T>();
+                if (typeof(T) == typeof(nuint))
```

Comment:
@tannergooding, the existing implementation of this method has separate specialization for ulong vs long, uint vs int, etc. Is that needed? Would it behave any differently if for long/ulong it just always AsUInt64 for example?

Reply:
Should be fine to mostly just do it based on size, it's mostly the same instruction for all of them anyways.

On some hardware there can be perf or execution port difference between `floating-point` vs `integer` (difference instructions), so it can be beneficial to preserve that difference, however.